CefSharp/Internals/JavascriptTypeBinder.cs

```diff
@@ -0,0 +1,398 @@
+        /// <summary>
+        /// Explicit Numeric Conversions Table
+        /// </summary>
+        static Dictionary<Type, Dictionary<Type, IConvert>> explicitConversions = new Dictionary<Type, Dictionary<Type, IConvert>>();
+
+        static JavascriptTypeBinder()
+        {
+            implicitConversions[TypeCode.SByte] = new TypeCode[] {
+                TypeCode.Int16,
+                TypeCode.Int32,
+                TypeCode.Int64,
+                TypeCode.Single,
+                TypeCode.Double,
+            };
+            implicitConversions[TypeCode.Byte] = new TypeCode[] {
+                TypeCode.Int16,
+                TypeCode.UInt16,
+                TypeCode.Int32,
+                TypeCode.UInt32,
+                TypeCode.Int64,
+                TypeCode.UInt64,
+                TypeCode.Single,
+                TypeCode.Double,
+                TypeCode.Decimal,
+            };
+            implicitConversions[TypeCode.Int16] = new TypeCode[] {
+                TypeCode.Int32,
+                TypeCode.Int64,
+                TypeCode.Single,
+                TypeCode.Double,
+                TypeCode.Decimal,
+            };
+            implicitConversions[TypeCode.UInt16] = new TypeCode[] {
+                TypeCode.Int32,
+                TypeCode.UInt32,
+                TypeCode.Int64,
+                TypeCode.UInt64,
+                TypeCode.Single,
+                TypeCode.Double,
+                TypeCode.Decimal,
+            };
+            implicitConversions[TypeCode.Int32] = new TypeCode[] {
+                TypeCode.Int64,
+                TypeCode.Single,
+                TypeCode.Double,
+                TypeCode.Decimal,
+            };
+            implicitConversions[TypeCode.UInt32] = new TypeCode[] {
+                TypeCode.Int64,
+                TypeCode.UInt64,
+                TypeCode.Single,
+                TypeCode.Double,
+                TypeCode.Decimal,
+            };
+            implicitConversions[TypeCode.Int64] = new TypeCode[] {
+                TypeCode.Single,
+                TypeCode.Double,
+                TypeCode.Decimal,
+            };
+            implicitConversions[TypeCode.Char] = new TypeCode[] {
+                TypeCode.UInt16,
+                TypeCode.Int32,
+                TypeCode.UInt32,
+                TypeCode.Int64,
+                TypeCode.UInt64,
+                TypeCode.Single,
+                TypeCode.Double,
+                TypeCode.Decimal,
+            };
+            implicitConversions[TypeCode.Single] = new TypeCode[] {
+                TypeCode.Double,
+            };
+            implicitConversions[TypeCode.UInt64] = new TypeCode[] {
+                TypeCode.Single,
+                TypeCode.Double,
+                TypeCode.Decimal,
+            };
+
+            // signed to unsigned conversion to silence overflow exception
+            new Converter<int, byte>((v) => (byte) v);
```

Comment:
Hmm, the new instances of `Converter` assign themselves to the dictionary. Very indirect approach, a more direct approach would be cleaner and easier I think.